app/src/main/java/openfoodfacts/github/scrachx/openfood/features/PreferencesFragment.kt

```diff
@@ -150,6 +151,37 @@ class PreferencesFragment : PreferenceFragmentCompat(), INavigationItem, OnShare
             true
         }
 
+        requirePreference<SwitchPreference>(getString(R.string.pref_scanner_type_key)).let {
+            it.isVisible = ContinuousScanActivity.showSelectScannerPref
+            it.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, newValue ->
+                if(newValue == true){
+                    MaterialDialog.Builder(requireActivity()).run {
+                        title(R.string.preference_choose_scanner_dialog_title)
+                        content(R.string.preference_choose_scanner_dialog_body)
+                        positiveText(R.string.proceed)
+                        onPositive { _, _ ->
+                            it.isChecked = true
+                            settings.edit { putBoolean(getString(R.string.pref_scanner_type_key), newValue as Boolean) }
+                            Toast.makeText(requireActivity(), getString(R.string.changes_saved), Toast.LENGTH_SHORT).show()
+                        }
+                        negativeText(R.string.dialog_cancel)
+                        onNegative {
+                            dialog, _ -> dialog.dismiss()
+                            it.isChecked = false
+                            settings.edit { putBoolean(getString(R.string.pref_scanner_type_key), false) }
+                        }
+                        show()
+                    }
+                }
+                else{
+                    it.isChecked = false
+                    settings.edit { putBoolean(getString(R.string.pref_scanner_type_key), newValue as Boolean) }
+                    Toast.makeText(requireActivity(), getString(R.string.changes_saved), Toast.LENGTH_SHORT).show()
```

Comment:
Can we use a snackbar?

Reply:
@VaiTon Shall I change Toast to Snackbar in the entire file?

Reply:
Yes please, let's not overuse Toast when we can.

Reply:
@VaiTon From what I figured yet, it is better to leave Toast in Preferences file. When we use ```listView``` (parent layout in Preferences ) as view parameter in Snackbar.make() , we get the following layout

![image](https://user-images.githubusercontent.com/71206228/110348554-c923aa80-8057-11eb-9525-273667165fd5.png)
i.e. it hides the bottom navigation bar. Is that alright?

AFAIK To view the snackbar above bottom navigation bar we will have to change the preferences.xml layout hierarchy so that we can add a CoordinatorLayout or something else to pass as view. What should we go for?

Reply:
@kartikaysharma01 could you push the changes so that I could see why it's not working as it should

Reply:
@VaiTon I have pushed the change where I used listview

Reply:
@kartikaysharma01 let's do it with Toast for now